lib/src/platform/platform.dart

```diff
@@ -0,0 +1,125 @@
+import 'package:logger/logger.dart';
+
+import 'platform_standard.dart'
+    if (dart.library.html) 'platform_web.dart'
+    if (dart.library.io) 'platform_vm.dart';
+
+typedef ConsolePrinter = void Function(String arg);
+
+/// Runtime platform capabilities
+class LogPlatform {
+  /// Returns [true] if running in Web platform (Browser).
+  static bool get isWeb => isPlatformWebImpl();
+
+  /// Returns [true] if running in VM platform.
+  static bool get isVM => isPlatformVMImpl();
+
+  /// Returns [true] if the platform console supports colors.
+  static bool get consoleSupportsColors => getPlatformConsoleAcceptsColorsImpl();
+
+  /// Returns console columns size.
+  static int get consoleColumns => getPlatformConsoleColumnsImpl();
+
+  /// Returns [true] if the console supports emoji.
+  static bool get consoleSupportsEmoji => isPlatformWebImpl();
+
+  static bool _enableSTDERR = false;
+
+  /// If set to [true], allows write of [Level.error] and
+  /// [Level.wtf] to `stderr` (if present in the runtime platform).
+  ///
+  /// Default to [false], to be backward compatible with previous versions.
+  /// of this package.
+  static bool get enableSTDERR => _enableSTDERR;
+
+  static set enableSTDERR(bool value) {
+    _enableSTDERR = value ?? false;
```

Comment:
Only ensuring that `_enableSTDERR` is always true/false, and `false` is the default value.

Reply:
Sure enough, but an engineer setting this field to `null` should face the music sooner rather than later. Clearly he didn't intent to do so since it makes little sense to do so. I'll concede that there might even be a valid use case in which you set this field to null. Then it's up to the caller to deal with the consequences.

The getter and setter can be replaced by a simple static field.